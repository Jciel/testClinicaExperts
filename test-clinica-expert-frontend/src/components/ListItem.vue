<template>
  <div class="list-item">
    <font-awesome-icon class="dot" icon="fa-solid fa-ellipsis-vertical" size="lg" />

    <div class="text-link">
      <span class="link-name">{{ shortLink.identifier }}</span>
      <span @click="emits('access', shortLink.identifier)" class="short-link" href="">{{ shortLink.urlShort }}</span>
    </div>

    <div class="info-actions-group">
      <div class="hits-stats">
        {{ shortLink.hits }}
        <font-awesome-icon class="stats" icon="fa-chart-simple" size="lg" />
      </div>

      <div class="actions-container">
        <font-awesome-icon class="icon-action" icon="fa-clone" rotation=90 size="lg" />
        <font-awesome-icon class="icon-action" icon="fa-pen-to-square" size="lg" />
        <font-awesome-icon @click="emits('delete', shortLink.id)" class="icon-action delete" icon="fa-trash-can" size="lg" />
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
  import { PropType } from "vue";
  import { ShortLink } from "../utils/types";

  const emits = defineEmits(['delete', 'access'])

  defineProps({
    shortLink: {
      type: Object as PropType<ShortLink>,
      required: true
    }
  })
</script>

<style scoped>
.list-item {
  border: 1px solid var(--color-borders);
  display: flex;
  border-radius: 5px;
  padding: 25px 30px;
  align-items: center;
  width: 100%;
  background-color: #fff;
  box-sizing: border-box;

  .dot {
    margin-right: 20px;
    color: var(--color-icon-info)
  }

  .text-link {
    display: flex;
    flex-direction: column;

    .short-link {
      cursor: pointer;
    }
  }

  .info-actions-group {
    display: flex;
    flex-direction: row;
    margin-left: auto;

    .hits-stats {
      margin-right: 30px;

      .stats {
        color: var(--color-icon-info)
      }
    }

    .actions-container {
      .icon-action {
        cursor: pointer;
        margin-right: 10px;
        color: var(--color-icon-action);

        &:last-child {
          margin-right: 0;
        }
      }
    }
  }
}
</style>
