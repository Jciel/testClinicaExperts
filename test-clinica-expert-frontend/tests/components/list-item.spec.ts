import {describe, expect, test} from "vitest"
import ListItem from "../../src/components/ListItem.vue";
import {mount} from "@vue/test-utils";


describe("Test ListItem.vue component", () => {
    test('mount component', async () => {
        expect(ListItem).toBeTruthy()

        const wrapper = mount(ListItem, {
            props: {
                shortLink: {
                    name: "shortLinkUrl",
                    url: "www.google.com.br",
                    shortUrl: "gg",
                    hits: 23
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot()

        const linkName = wrapper.find('.link-name')
        const shortLinkUrl = wrapper.find('.short-link')

        expect(linkName.text()).toStrictEqual('shortLinkUrl')
        expect(shortLinkUrl.text()).toStrictEqual('gg')
    })

    test('test if emit a delete event when delete icon is clicled', async () => {
        const wrapper = mount(ListItem, {
            props: {
                shortLink: {
                    name: "shortLinkUrl",
                    url: "www.google.com.br",
                    shortUrl: "gg",
                    hits: 23
                }
            }
        })

        await wrapper.find('.delete').trigger('click')
        expect(wrapper.emitted('delete'))
    })
})
