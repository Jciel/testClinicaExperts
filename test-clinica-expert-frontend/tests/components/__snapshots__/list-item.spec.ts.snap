// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test ListItem.vue component > mount component 1`] = `
"<div data-v-ee8f0cc4=\\"\\" class=\\"list-item\\">
  <font-awesome-icon data-v-ee8f0cc4=\\"\\" class=\\"dot\\" icon=\\"fa-solid fa-ellipsis-vertical\\" size=\\"lg\\"></font-awesome-icon>
  <div data-v-ee8f0cc4=\\"\\" class=\\"text-link\\"><span data-v-ee8f0cc4=\\"\\" class=\\"link-name\\">shortLinkUrl</span><span data-v-ee8f0cc4=\\"\\" class=\\"short-link\\" href=\\"\\">gg</span></div>
  <div data-v-ee8f0cc4=\\"\\" class=\\"info-actions-group\\">
    <div data-v-ee8f0cc4=\\"\\" class=\\"hits-stats\\">23 <font-awesome-icon data-v-ee8f0cc4=\\"\\" class=\\"stats\\" icon=\\"fa-chart-simple\\" size=\\"lg\\"></font-awesome-icon>
    </div>
    <div data-v-ee8f0cc4=\\"\\" class=\\"actions-container\\">
      <font-awesome-icon data-v-ee8f0cc4=\\"\\" class=\\"icon-action\\" icon=\\"fa-clone\\" rotation=\\"90\\" size=\\"lg\\"></font-awesome-icon>
      <font-awesome-icon data-v-ee8f0cc4=\\"\\" class=\\"icon-action\\" icon=\\"fa-pen-to-square\\" size=\\"lg\\"></font-awesome-icon>
      <font-awesome-icon data-v-ee8f0cc4=\\"\\" class=\\"icon-action delete\\" icon=\\"fa-trash-can\\" size=\\"lg\\"></font-awesome-icon>
    </div>
  </div>
</div>"
`;
